components/claim-mgt/org.wso2.carbon.identity.claim.metadata.mgt/src/main/java/org/wso2/carbon/identity/claim/metadata/mgt/dao/LocalClaimDAO.java

```diff
@@ -197,8 +198,15 @@ private void addClaimAttributeMappings(Connection connection, int localClaimId,
             try {
                 String query = SQLConstants.ADD_CLAIM_MAPPED_ATTRIBUTE;
                 prepStmt = connection.prepareStatement(query);
-
+                if (attributeMappings.isEmpty()) {
```

Comment:
Would not it better to skip execution if attribute mapping not present, rather than throwing exception?

Reply:
@ruwanta What is the best way to skip the execution?

Reply:
if (!attributeMappings.isEmpty()) {
 for(..) {..}
}
I do not see reason us throw an exception when attributeMappings list is empty. Perhaps null check is what we want.

also we need to move "prepStmt = connection.prepareStatement(query);"  after the if check. No point preparing something which is not going to be used.